modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/Processing.scala

```diff
@@ -78,13 +84,15 @@ object Processing {
     }
   }
 
-  private def dropViews[F[_]: Sync](env: Environment[F], ref: Ref[F, WindowState]): F[Unit] =
-    ref.get.flatMap { state =>
-      if (state.framesOnDisk.nonEmpty)
-        env.cpuPermit.surround(env.lakeWriter.removeDataFramesFromDisk(state.framesOnDisk))
-      else
-        Sync[F].unit
-    }
+  /**
+   * Manages the lifecycle of initializing and cleaning a Spark DataFrame scoped to the lifetime of
+   * the window
+   */
+  private def manageDataFrame[F[_]](env: Environment[F], viewName: String): Stream[F, Unit] = {
```

Comment:
Another idea: how about `WindowState.build` returns `Resource` and we manage init/close there? Then simple `Stream.resource`, maybe this would a bit more explicit what's our window lifecycle?